### DISCUSSION TITLE
the number of time steps in postprocessing

### DISCUSSION THREAD

**Discussion by snugook1108** [idaholab/moose#17996] [Q&A General] [answered]
Hi, did you know which type/variable setup for the number of time steps in postprocessing?

**Comment by GiudGiud** [comment, marked as answer]
Hello

Not sure I fully understand your question. Postprocessing is done by default at the end of every timestep in moose. This can be controlled using the `execute_on` parameter to most objects. You can make a custom `execute_on` option that does the postprocessing only when you wish. Also please note the `Output` objects are also controlled with this system.

 Maybe look at https://mooseframework.inl.gov/source/interfaces/SetupInterface.html which has more information on how that system works. 

Best,
Guillaume

**Reply by aeslaughter** [reply]
Just to clarify the output side: the "execute_on" in an Output object (i.e, [Outputs] in your input file) dictates when an the Output object is executed, it has no control on when the information being computed occurs.

https://mooseframework.inl.gov/syntax/Outputs/index.html
